media/js/base/search-params.js

```diff
@@ -81,5 +81,11 @@ _SearchParams.prototype.utmParamsFxA = function(pathname) {
 
     /* eslint-enable camelcase */
 
+    // ensure all values are strings, as no numeric values are allowed
+    // into UITour.showFirefoxAccounts
+    Object.keys(utms).forEach(function(key) {
+        utms[key] = utms[key].toString();
```

Comment:
Could you avoid enumerating through the keys twice if you did `utms[key] = val.toString();` on line 51 inside `_SearchParams.prototype.utmParams`?

Reply:
Since `_SearchParams.prototype.utmParams` is also used by our stub attribution script, I was trying to avoid regressions there (as I'm not super familiar and this bug is currently causing issues in prod). Also, I'm not sure it's "correct" to force strings for all values all the time - FxA seems to be the only place strings are required.

What do you think?

Reply:
Hmm, ok I guess? I thought `toString()` should have been pretty safe, but I guess we could always err on the side of caution? How confident are we that this function will only ever get called via a browser that supports `forEach` on an Array?

Reply:
Pretty confident? It should only be called for Firefox, right? Happy to go back to an earlier solution if you see gaps in my logic though.

Reply:
Good enough for me 🍡